aten/src/ATen/native/cuda/AmpKernels.cu

```diff
@@ -0,0 +1,103 @@
+#define _USE_MATH_DEFINES
+
+#include <math.h>
+
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cuda/Loops.cuh>
+
+
+namespace at {
+namespace native {
+
+// Multiplies scaled_grad in-place by rscale.  If an element of scaled_grad was inf or NaN sets found_inf to 1.
+//
+// Args:
+// scaled_grad:  A (scaled) gradient tensor.  May contain infs or NaNs.
+// rscale:  The inverse of the scale factor by which scaled_grad is currently multiplied.
+// found_inf:  A single-element float tensor to which 1.0 will be written if any gradients contain infs/nans.
+//             Pre-zeroing found_inf, if appropriate, is the responsibility of the caller.
+// Returns:
+// A reference to the grad, which was unscaled in place.
+Tensor& _amp_unscale_inf_check_cuda(Tensor& scaled_grad,
+                                     const Tensor& rscale,
+                                     const Tensor& found_inf)
+{
+  TORCH_CHECK(scaled_grad.is_cuda(), "scaled_grad must be a CUDA tensor.");
+  TORCH_CHECK(rscale.is_cuda(), "rscale must be a CUDA tensor.");
+  TORCH_CHECK(found_inf.is_cuda(), "found_inf must be a CUDA tensor.");
+  TORCH_CHECK(rscale.numel() == 1, "rscale must be a 1-element tensor.");
+  TORCH_CHECK(found_inf.numel() == 1, "found_inf must be a 1-element tensor.");
+  TORCH_CHECK(rscale.scalar_type() == at::ScalarType::Float, "rscale must be a float tensor.");
+  TORCH_CHECK(found_inf.scalar_type() == at::ScalarType::Float, "found_inf must be a float tensor.");
+
+  // Act on scaled_grad in place.
+  auto iter = TensorIterator::unary_op(scaled_grad, scaled_grad);
+
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(
+    iter.dtype(),
+    "_amp_unscale_inf_check_kernel",
+    [&] {
+      auto* found_inf_ptr = found_inf.data_ptr<float>();
+      auto* rscale_ptr = rscale.data_ptr<float>();
+
+      gpu_kernel(iter, [found_inf_ptr, rscale_ptr] GPU_LAMBDA(scalar_t val) -> scalar_t {
+          auto fval = static_cast<float>(val);
+          if(!std::isfinite(fval))
```

Comment:
All conditionals should have curly braces to make the linter happy. (Also a space between the if an parens would be nice.)

if (...) { 
  ...
}